### PR TITLE
Change spawnning of robots during import

### LINKED ISSUE #597: ROS 2 Spawn point and ROS 2 Robot Importer
**Describe the bug**
There are two issues with the spawn point.
- the importer should propose a default spawn point instead of the first one:
[Screencast from 10-31-2023 05:33:18 PM.webm](https://github.com/o3de/o3de-extras/assets/3209244/7950d7ba-5cd1-460a-99cb-dfa2204bcf4a)
- the prefab should be moved to the spawn point, instead of the entity in the prefab:
image
Prefab is in origin.

**Assets required**
Any URDF.

**Steps to reproduce**
1. Add some spawn points and move them around
2. Import robot,

**Expected behavior**
- The default spawn point should be "default"
- Prefab should be instantiated at spawn point not the base_link

**Found in Branch**
o3de-extras/develepoment.

### PR DESCRIPTION
**Pull request by arturkamieniecki**
I've modified the spawning behavior of the Robot Importer to always display the default spawn position as the first one. I've also changed the importer such that robot prefabs are moved and not the first entity during the move.

Resolves #597

### CODE REVIEW

File: Gems/ROS2/Code/Source/RobotImporter/Pages/PrefabMakerPage.cpp [hunk 1]
```diff
@@ -34,9 +34,11 @@ namespace ROS2
         m_spawnPointsComboBox = new QComboBox(this);
         m_spawnPointsInfos = allActiveSpawnPoints.values;
 
-        for (int i = 0; i < allActiveSpawnPoints.values.size(); i++)
+        m_spawnPointsComboBox->addItem("default", QVariant(0));
+
+        for (int i = 1; i <= allActiveSpawnPoints.values.size(); i++)
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/RobotImporter/Pages/PrefabMakerPage.cpp:39]: Is this about 0- vs 1-indexing? If so, add a comment please.

File: Gems/ROS2/Code/Source/RobotImporter/Pages/PrefabMakerPage.cpp [hunk 2]
```diff
@@ -96,8 +98,12 @@ namespace ROS2
         if (!m_spawnPointsInfos.empty())
         {
             int vectorIndex = m_spawnPointsComboBox->currentData().toInt();
+            if (vectorIndex == 0)
+            {
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/RobotImporter/Pages/PrefabMakerPage.cpp:102]: Add a comment please why this is a bad value

File: Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp [hunk 3]
```diff
@@ -429,6 +428,10 @@ namespace ROS2
             auto prefabInterface = AZ::Interface<AzToolsFramework::Prefab::PrefabPublicInterface>::Get();
             [[maybe_unused]] auto createPrefabOutcome =
                 prefabInterface->InstantiatePrefab(relativePath.String(), AZ::EntityId(), AZ::Vector3::CreateZero());
+            if (createPrefabOutcome.IsSuccess())
+            {
+                MoveEntityToDefaultSpawnPoint(createPrefabOutcome.GetValue(), m_spawnPosition);
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp:433]: User selects the spawn point out of a combo box when using the importer - it probably should not be named "default" in such case.

File: Gems/ROS2/Code/Source/RobotImporter/Pages/PrefabMakerPage.cpp (hunk 2 above)
Comment (michalpelka) [reply Gems/ROS2/Code/Source/RobotImporter/Pages/PrefabMakerPage.cpp:102]: Please implement IsDefault method to make it more clear and use constexpr char* DefaultSpawnPointLabel =  "default".

File: Gems/ROS2/Code/Source/RobotImporter/Pages/PrefabMakerPage.cpp [hunk 4]
```diff
@@ -34,9 +34,11 @@ namespace ROS2
         m_spawnPointsComboBox = new QComboBox(this);
         m_spawnPointsInfos = allActiveSpawnPoints.values;
 
-        for (int i = 0; i < allActiveSpawnPoints.values.size(); i++)
+        m_spawnPointsComboBox->addItem("default", QVariant(0));
```
Comment (michalpelka) [inline Gems/ROS2/Code/Source/RobotImporter/Pages/PrefabMakerPage.cpp:37]: Move "default" to constexpr and reuse it in isDefault method.
Do not forget about i18n and _tr("") macro in Qt code.

**Review by michalpelka — COMMENTED** [review]
I've tested, work as intended, please adjust to review.

File: Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp (hunk 3 above)
Comment (arturkamieniecki) [reply Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp:433]: There are two positions that can be called default. One is the 0,0,0 point. I've changed the name of it to simulation origin. The second is the default spawning position of the spawn point. In the component, it is called default and can be other than 0,0,0.

**Comment by arturkamieniecki** [comment]
The new selection: 
Screenshot from 2023-11-07 16-26-23

File: Gems/ROS2/Code/Source/RobotImporter/Pages/PrefabMakerPage.h [hunk 5]
```diff
@@ -41,12 +41,16 @@ namespace ROS2
         void onCreateButtonPressed();
 
     private:
+        static bool IsZeroPoint(AZStd::string spawnPointName);
+
+        static constexpr AZStd::string_view zeroPoint = "Simulation origin";
```
Comment (michalpelka) [inline Gems/ROS2/Code/Source/RobotImporter/Pages/PrefabMakerPage.h:46]: It should be capitalized,

**Review by michalpelka — APPROVED** [review]

**Pull request merged**